config/m4/compiler.m4

```diff
@@ -214,15 +214,15 @@ AC_DEFUN([CHECK_COMPILER_FLAG],
          SAVE_CXXFLAGS="$CFLAGS"
          CFLAGS="$BASE_CFLAGS $CFLAGS $2"
          CXXFLAGS="$BASE_CXXFLAGS $CXXFLAGS $2"
-         AC_COMPILE_IFELSE([$3],
-                           [AC_MSG_RESULT([yes])
-                            CFLAGS="$SAVE_CFLAGS"
-                            CXXFLAGS="$SAVE_CXXFLAGS"
-                            $4],
-                           [AC_MSG_RESULT([no])
-                            CFLAGS="$SAVE_CFLAGS"
-                            CXXFLAGS="$SAVE_CXXFLAGS"
-                            $5])
+         AC_LINK_IFELSE([$3],
+                        [AC_MSG_RESULT([yes])
+                         CFLAGS="$SAVE_CFLAGS"
+                         CXXFLAGS="$SAVE_CXXFLAGS"
+                         $4],
+                        [AC_MSG_RESULT([no])
+                         CFLAGS="$SAVE_CFLAGS"
+                         CXXFLAGS="$SAVE_CXXFLAGS"
+                         $5])
```

Comment:
I am not sure how changing white-space here fixes the problem.

Reply:
@jtoppins the main change is AC_COMPILE_IFELSE -> AC_LINK_IFELSE. In the original build issue, it was eventually failing during link.
unfortunately, we cannot rely on `$ac_cv_c_compiler_gnu` since on non-gnu compilers [e.g pgi] this icc-specific check is also supposed to fail.

Reply:
I see its the AC_COMPILE_IFELSE vs AC_LINK_IFELSE

Reply:
Ok looks good to me.